openff/toolkit/utils/rdkit_wrapper.py

```diff
@@ -888,6 +889,10 @@ def generate_conformers(
             randomSeed=1,
             # params=AllChem.ETKDG()
         )
+
+        if rdmol.GetNumConformers() == 0:
+            raise ConformerGenerationError("RDKit conformer generation failed.")
```

Comment:
Does RDKit output any of its own error messages?

Reply:
Not as far as I'm aware, but I don't use RDKit directly too often so I could be missing something.

```python3
>>> from rdkit import Chem
>>> from rdkit.Chem import AllChem
>>> rdmol = Chem.MolFromSmiles(100 * "CC")
>>> rdmol = Chem.AddHs(rdmol)
>>> AllChem.EmbedMultipleConfs(rdmol, numConfs=1, pruneRmsThresh=1)
<rdkit.rdBase._vecti object at 0x13763e200>
>>> rdmol.GetNumConformers()
0
```

Reply:
That's disappointing :(.

Maybe you could query the output directly? It's supposed to contain the conformer IDs of the conformers generated, so if there aren't any, that should correspond to conformer generation failing. Conformers are already cleared by default (`clearConfs=True`) so this should lead to the same behaviour.

```python
status = AllChem.EmbedMultipleConfs(
            rdmol,
            numConfs=n_conformers,
            pruneRmsThresh=rms_cutoff / unit.angstrom,
            randomSeed=1,
            # params=AllChem.ETKDG()
        )
if not status:
    raise ConformerGenerationError("RDKit conformer generation failed.")
```

Incidentally, I do get a conformer if I start off with random coordinates. I wonder if the distance matrix is having difficulty with the number of atoms.

```python
>>> status = AllChem.EmbedMultipleConfs(rdmol, numConfs=1, useRandomCoords=True)
>>> list(status)
[0]
```

Looks alright I guess.

<img width="591" alt="Screen Shot 2021-08-24 at 9 46 19 am" src="https://user-images.githubusercontent.com/31115101/130656890-0df3ac2a-e525-4834-8410-1ccdb24dafe2.png">

Reply:
Looks like a reasonable C200 to me.

I don't think this is a good way of testing for this failures, since other settings or future RDKit updates could make generating stupidly large molecules safer/faster, and I'd like to avoid asking it to do an even larger molecule like a million-carbon alkane just to catch a failure. I figure there's got to be a small molecule out there that will reliably fail to be turned into 3D, but not for other reasons (i.e. RDKit wouldn't throw an error for other reasons). But that's as far as I got with that line of thinking. Any suggestions?

Reply:
One option would be to blindly grab something from their issue tracker, but that's more or less the opposite of maintainable code. The C200 I have right now fails in about 50 seconds on my machine, but two of the smaller molecules from https://github.com/rdkit/rdkit/issues/1433 fail in about 15 seconds each.

Reply:
No idea, sorry 😬 from that issue I'd guess it's flexible or large molecules that give it trouble, although I can't work out what's especially hard about `"C1CC1c2[nH]nc3C4CCC4c23"`. I did make a variation on a theme (`"C12C3C(C1C2)C4=N[N](C(=C34)C5C6C5CC6)C7CC7"`) for something that might not get directly addressed in the future.

![image](https://user-images.githubusercontent.com/31115101/130661316-83b5ee76-77aa-4ea0-b7a2-b7e2d707aed7.png)

```python
>>> rdmol = Chem.MolFromSmiles("C12C3C(C1C2)C4=N[N](C(=C34)C5C6C5CC6)C7CC7")
>>> rdmol = Chem.AddHs(rdmol)
>>> status = AllChem.EmbedMultipleConfs(rdmol, numConfs=1, useRandomCoords=True)
>>> len(status)
0
```

Reply:
I asked in Slack and the PIs chimed in with a list of suggestions, the smallest of which was uranium hexafluoride, which I'm rolling with. Coincidentally it has the funniest SMILIES string I've seen to date ...